### PR TITLE
Fix unit tests using the database

### LINKED ISSUE #304: Make unit tests use separate database

### PR DESCRIPTION
**Pull request by sam-eng**
resolves #304

### CODE REVIEW

**Comment by sam-eng** [comment]
Note: For some reason TravisCI is failing on because of something with bcrypt... I'm going to look into this further and see if replacing bcrypt with bcryptjs will fix things. Not sure why this is happening...

**Comment by sam-eng** [comment]
> Note: For some reason TravisCI is failing on because of something with bcrypt... I'm going to look into this further and see if replacing bcrypt with bcryptjs will fix things. Not sure why this is happening...

Fixed this by replacing bcrypt with bcryptjs, this shouldn't affect much--just everywhere in the code where it says "bcrypt" make sure it gets replaced with "bcryptjs"--nothing else should need to be changed.

File: back-end/models/User.js [hunk 1]
```diff
@@ -21,15 +21,13 @@ UserSchema.methods.validPassword = async function (password) {
...
+  const hash = await bcryptjs.hash(this.password, saltRounds)
+  this.password = hash
   next()
 })
 
-const User = mongoose.model('user',UserSchema)
+const User = mongoose.model('user', UserSchema)
 
 module.exports = User
-
```
Comment (almazhankapan) [inline back-end/models/User.js:34]: it seems that there is one bcrypt left inside the validPassword function

**Review by almazhankapan — APPROVED** [review]

**Comment by almazhankapan** [comment]
Great job!

**Pull request merged**